tests/integration/test_match_merger.py

```diff
@@ -27,6 +29,4 @@ def test_match_merger(docs, shards):
         documents = f.search(docs, return_results=True)[0].docs
         assert len(documents) == 2
         for doc in documents:
-            assert {d.tags['shard_id'] for d in doc.matches} == {
-                float(i) for i in range(shards)
-            }
+            assert [d.scores['cosine'].value for d in doc.matches] == [shards - 1]
```

Comment:
This assert looks weird? are we comparing a list to a list with a single value?

Reply:
Sorry, it is not very clear at first glance. This basically means that after `MatchMerger`, all the matches from other shards are removed except the match from the shard which has the highest (score)cosine-value. In this case the cosine values are just shard numbers. So, only the match from the last shard would be left.

Maybe, I can add some more matches in the matcharray to make it clearer.

<hr/>

For the above comment, would it make sense to retrieve the metric-name via parameters dict. If it is not present, then just use cosine. 
```
metric_name = parameters.get('metric_name', 'cosine')
```
The only problem is when the default value for metric_name is changed in `DocumentArrayNeuralOpsMixin` at some later point of time.

Thanks :)

Reply:
For the first concern, let's put readability before speed to make everyone understands, but as @maximilianwerk said, the logic may be broken.

For the second, what I would do is:

`set metric_name` as attribute in `__init__`, and then read the metric_name from parameters defaulting to `self._metric_name` as `metric_name = parameters.get('metric_name', self._metric_name)`